source/manage_the_buzz_for_developers/applications_and_user_system_manage_the_buzz/custom_solution_for_self_managed_user_system/introduction_to_service_apps.rst

```diff
@@ -2,10 +2,14 @@
 
 Introduction to Service Apps
 ============================
+Applications that have long-running processes or that operate without interaction with a user also need a way to access secured resources.
+For this OAuth 2.0 Client Credentials grant flow is used.
 
-.. figure:: /_static/coming-soon.png
-   :align: center
+They are used in the following scenarios:
 
-   **Coming Soon**
+1. Where you are providing your own user management system and bypass the COCO User management system
+2. Where you are developing an application or integration that does not require authorization from a specific COCO user.
+
+.. image:: ../../../_static/serviceAppFlow.png
```

Comment:
At the end of the flow, isn't there a call to the COCO Server missing to refresh the access token?

Reply:
Added in the flow

Reply:
At the end, the fetch-token call and its response are repeated twice, and the setToken call is in the wrong place. Also, the last response call is missing the body of the response containing the new token. The order of steps at the end should match the order of steps at the beginning and none of the steps should be repeated twice, and the body of the request or response should be included wherever it is applicable:

[After oauthcallback]
1. fetch-token call from Third Party App to Third Party Server
2. .../token call to COCO Server
3. Response from COCO Server to Third Party Server
4. Response from Third Party Server to Third Party App
5. setToken call from Third Party App to COCO SDK

Reply:
Also - change the swimlane name "COCO SDK" to "COCO Client SDK / COCO Media SDK"

Reply:
Updated